server/sonar-web/src/main/js/apps/projects/components/DefaultPageSelector.js

```diff
@@ -23,23 +23,65 @@ import { connect } from 'react-redux';
 import { withRouter } from 'react-router';
 import AllProjectsContainer from './AllProjectsContainer';
 import { getCurrentUser } from '../../../store/rootReducer';
-import { shouldRedirectToFavorite } from '../utils';
+import { isFavoriteSet, isAllSet } from '../utils';
+import { searchProjects } from '../../../api/components';
+
+type Props = {
+  currentUser: { isLoggedIn: boolean },
+  location: { query: {} },
+  router: { replace: (path: string) => void }
+};
+
+type State = {
+  shouldByRedirected?: boolean
```

Comment:
You mean `shouldBeRedirected` no ? And not `shouldBy...` ? It might be confusing when using later on no ?